test/src/unittests/rhythm/test_loopbpmconfidence.py

```diff
@@ -0,0 +1,157 @@
+         Test for different BPMs starting with correct one(125),
+         and check the confidence levels obtained from previous runs.
+         The 3rd param in assertAlmostEqual() has been optimised to 0.02.
+        """        
+
+        bpmEstimate = 125
+        expectedConfidence = 1
+        confidence = LoopBpmConfidence()(audio, bpmEstimate)        
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.02)
+
+        bpmEstimate = 120
+        expectedConfidence = 0.87
+        confidence = LoopBpmConfidence()(audio, bpmEstimate)        
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.02)
+
+        bpmEstimate = 70
+        expectedConfidence = 0.68
+        confidence = LoopBpmConfidence()(audio, bpmEstimate)                
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.02)
+
+        """
+         Apply same principle above to Sub-Sections of Audio.       
+         The following confidence measurements were taken on 11 Feb. 2021
+         for a given set of parameters on different audio subsections.
+         A segment of techno_loop gives a non-ideal estimation of BPM.
+         
+         Subsection [1000:6000], (Randomly chosen points, section 1)
+         BPM Estimate: 125: Confidence: 0.5275888442993164
+         BPM Estimate: 120: Confidence: 0.5464853048324585
+         BPM Estimate: 70: Confidence: 0.7354497313499451
+         Subsection [22000:38000], (Randomly chosen points, section 2)
+         BPM Estimate: 125: Confidence: 0.5117157697677612
+         BPM Estimate: 120: Confidence: 0.4512471556663513
+         BPM Estimate: 70: Confidence: 0.15343916416168213
+         The ASSERTs will check for these to two places of decimal
+        
+         These checks act as integrity checks. If any future changes break these asserts,
+         then this will indicate that accuracy of algorithm has changed.
+         The 3rd param in assertAlmostEqual() has been optimised, lowest possible value chosen.
+        """
+
+        # Randomly chosen points, Section 1
+        samplePoint1 = 1000
+        samplePoint2 = 6000
+
+        bpmEstimate = 125
+        expectedConfidence = 0.53
+        confidence = LoopBpmConfidence()(audio[samplePoint1:samplePoint2], bpmEstimate)                
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.02) 
+
+        bpmEstimate = 120
+        expectedConfidence = 0.55
+        confidence = LoopBpmConfidence()(audio[samplePoint1:samplePoint2], bpmEstimate)                
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.02) 
+        
+        bpmEstimate = 70
+        expectedConfidence = 0.74
+        confidence = LoopBpmConfidence()(audio[samplePoint1:samplePoint2], bpmEstimate)            
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.02) 
+
+        # Randomly chosen points, Section 2
+        samplePoint1 = 22000
+        samplePoint2 = 38000
+        expectedEstimate = 100
```

Comment:
This accounts for 0.36 seconds. This is too short (see the previous comment) as we want something that is long enough to account for a large segment of a 4/4 bar.